src/common/commonutils/SshUtils.c

```diff
@@ -220,171 +235,180 @@ int CheckAppropriateCiphersForSsh(const char** ciphers, unsigned int numberOfCip
 
     return status;
 }
 
-int CheckSshOptionIsSetToString(const char* option, const char* expectedValue, char** reason, void* log)
+static int CheckSshOptionIsSetToInteger(const char* option, int* expectedValue, int* actualValue, char** reason, void* log)
 {
-    char* value = NULL;
-    int status = 0;
+    char* actualValueString = NULL;
+    char* expectedValueString = expectedValue ? FormatAllocateString("%d", *expectedValue) : NULL;
+    int status = CheckSshOptionIsSet(option, expectedValueString, &actualValueString, reason, log);
 
-    if ((NULL == option) || (NULL == expectedValue))
+    if ((0 == status) && (NULL != actualValue))
     {
-        OsConfigLogError(log, "CheckSshOptionIsSetToString: invalid arguments (%s, %s)", option, expectedValue);
-        return EINVAL;
+        *actualValue = (NULL != actualValueString) ? atoi(actualValueString) : -1;
     }
 
-    if (false == IsSshServerActive(log))
+    FREE_MEMORY(actualValueString);
+    FREE_MEMORY(expectedValueString);
+
+    return status;
+}
+
+int CheckSshClientAliveInterval(char** reason, void* log)
+{
+    int actualValue = 0;
+    int status = CheckSshOptionIsSetToInteger("clientaliveinterval", NULL, &actualValue, reason, log);
+    
+    if ((0 == IsSshServerActive(log)) && (actualValue <= 0))
     {
-        OsConfigLogInfo(log, "CheckSshOptionIsSetToString: the SSH Server daemon is not active on this device");
-        return status;
+        OsConfigLogError(log, "CheckSshClientAliveInterval: 'clientaliveinterval' is not set to a greater than zero value in SSH Server response (but to %d)", actualValue);
+        OsConfigCaptureReason(reason, "'clientaliveinterval' is not set to a greater than zero value in SSH Server response (but to %d)",
+            "%s, also 'clientaliveinterval' is not set to a greater than zero value in SSH Server response (but to %d)", actualValue);
+        status = ENOENT;
     }
 
-    if (NULL != (value = GetSshServerState(option, log)))
-    {
-        OsConfigLogInfo(log, "CheckSshOptionIsSetToString: '%s' found in SSH Server response set to '%s'", option, value);
+    OsConfigLogInfo(log, "CheckSshClientAliveInterval: %s (%d)", status ? "failed" : "passed", status);
 
-        if (0 != strcmp(value, expectedValue))
-        {
-            OsConfigLogError(log, "CheckSshOptionIsSetToString: '%s' is not set to '%s' in SSH Server response (but to '%s')", option, expectedValue, value);
-            OsConfigCaptureReason(reason, "'%s' is not set to '%s' in SSH Server response (but to '%s')",
-                "%s, also '%s' is not set to '%s' in SSH Server response (but to '%s')", option, expectedValue, value);
-            status = ENOENT;
-        }
+    return status;
+}
 
-        FREE_MEMORY(value);
-    }
-    else
+int CheckSshLoginGraceTime(char** reason, void* log)
+{
+    int actualValue = 0;
+    int status = CheckSshOptionIsSetToInteger("logingracetime", NULL, &actualValue, reason, log);
```

Comment:
Should turn `logingracetime` into a constant

Reply:
Same as commented elsewhere.